GrassChoppers/Modules/Home/HomeDataManager.swift

```diff
@@ -1,13 +1,24 @@
 protocol HomeDataManaging: class {
     var delegate: HomeDataManagingDelegate? { get set }
+    
+    func requestInitialState()
 }
 
 protocol HomeDataManagingDelegate: class {
-    
+    func didFetch(name: String)
 }
 
 class HomeDataManager: HomeDataManaging {
     
     weak var delegate: HomeDataManagingDelegate?
     
+    func requestInitialState() {
+        print("Data manager got the message")
+        
+        // step 0: fetch data from remote database
+        
+        // step 1: notify logic controller
+        delegate?.didFetch(name: "Wasay")
+    }
```

Comment:
This function is the entrypoint for `HomeDataManager` to have its 'initial state' requested. LogicController will call this function when the view loads to request the data necessary to show on-launch.

Notice that it mocks a call to a database (over the internet, slow), and then sends the response to its delegate.
We can't just return the response from this function, because in the real world, the database server wont have our data back to us in time. We need an asynchronous way of returning the response. We use the delegate pattern to do this.

The delegate is just a property on `HomeDataManager` that hold a reference to some object (the logic controller). That delegate object is of type `HomeDataManagingDelegate`, meaning: the delegate type (class) must conform to the `HomeDataManagingDelegate` protocol. That way, we can be sure that the delegate has the functions that we need to send the response. Swift will ensure, when we conform to the protocol (`extension LogicController: HomeDataManagingDelegate`), that `LogicController` will have the same functions. Now `LogicController` is ready to accept the response.